engine/execution/ingestion/stop_at_height.go

```diff
@@ -0,0 +1,176 @@
+package ingestion
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type StopControl struct {
+	sync.RWMutex
+	height             uint64
+	crash              bool
+	commenced          bool // if stopping process has started. We disallow any changes then.
```

Comment:
Can we get rid of this state by:

```
isCommenced := s.stopAfterExecuting != flow.ZeroID
```

Reply:
Nope, this should be set to true if we skip any block, which include skipping blocks above the height, not necessary setting the final block ID.